modules/harvest/src/Commands/HarvestCommands.php

```diff
@@ -127,13 +127,16 @@ protected function buildPlanFromOpts($opts) {
    */
   public function deregister($id) {
     $message = 'Could not deregister the ' . $id . ' harvest.';
-    try {
+    $this->logger->warning(
+      'If you deregister a harvest with published datasets, you will
+       not be able to bulk revert the datasets connected to this harvest.');
+    if ($this->io()->confirm("Deregister harvest {$id}")) {
       if ($this->harvestService->deregisterHarvest($id)) {
         $message = 'Successfully deregistered the ' . $id . ' harvest.';
       }
     }
-    catch (\Exception $e) {
-      $message = $e->getMessage();
+    else {
+      $message = "Skipped deregistering harvest {$id}";
```

Comment:
Just a minor request @stefan-korn - "skipped" seems like the wrong phrasing since the user simply chose to abort the command. I think in other drush commands with a confirmation like this the standard is simply to throw a `Drush\Exceptions\UserAbortException`. 

See for instance https://github.com/drush-ops/drush/blob/12.x/src/Commands/core/SiteInstallCommands.php#L361

Reply:
Okay, I have changed this.